### PR TITLE
Path planner master

### PR DESCRIPTION
**Pull request by p3jawors**
Proposed restructuring of path planners

- not tested, but completed outline for second order path planners
- master class has generate_path_function and next_timestep functions
- subclasses have pregenerate_path, step, and plot_trajectory functions
- calling generate_path_function will call the subclasses
  pregenerate_path and will interpolate to the specified time. This can
  then be stepped through by calling next_timestep
- calling step in the subclass will work the same as before, except if
  the pregenerate_path function has been called, in which case it will
  pass the next target in that list

Opening for discussion, will complete for review afterwards

### CODE REVIEW

**Comment by p3jawors** [comment]
- style checks passing
- examples run
- ready for review

File: abr_control/arms/base_config.py [hunk 1]
```diff
@@ -299,6 +302,35 @@ def R(self, name, q):
         parameters = tuple(q)
         return np.array(self._R[name](*parameters), dtype='float32')
 
+    #TODO: optimize this function
+    def euler_angles(self, name, q):
```
Comment (studywolf) [inline abr_control/arms/base_config.py]: this function doesn't appear to be used at all in the code

File: abr_control/arms/base_config.py [hunk 2]
```diff
@@ -299,6 +302,35 @@ def R(self, name, q):
...
+
+        Parameters
+        ----------
+        name : string
+            name of the joint, link, or end-effector
+        q : numpy.array
+            joint angles [radians]
+        """
+        R = self.R(name=name, q=q)
+        euler = transformations.euler_from_matrix(matrix=R)
```
Comment (studywolf) [inline abr_control/arms/base_config.py]: need to specify the `axes` here for clarity

**Comment by p3jawors** [comment]
New scipy release for quaternion interpolation. Might be worth switching over.
https://scipy.github.io/devdocs/generated/scipy.spatial.transform.RotationSpline.html

**Comment by p3jawors** [comment]
Changes LGTM, examples run, style checks pass

**Pull request merged**